remoting/src/main/java/org/apache/rocketmq/remoting/RemotingClient.java

```diff
@@ -45,6 +47,30 @@ void invokeOneway(final String addr, final RemotingCommand request, final long t
         throws InterruptedException, RemotingConnectException, RemotingTooMuchRequestException,
         RemotingTimeoutException, RemotingSendRequestException;
 
+    default CompletableFuture<RemotingCommand> invoke(final String addr, final RemotingCommand request,
+        final long timeoutMillis) {
+        CompletableFuture<RemotingCommand> future = new CompletableFuture<>();
+        try {
+            invokeAsync(addr, request, timeoutMillis, responseFuture -> {
+                RemotingCommand response = responseFuture.getResponseCommand();
+                if (response != null) {
+                    future.complete(response);
+                } else {
+                    if (!responseFuture.isSendRequestOK()) {
+                        future.completeExceptionally(new RemotingSendRequestException(addr, responseFuture.getCause()));
+                    } else if (responseFuture.isTimeout()) {
+                        future.completeExceptionally(new RemotingTimeoutException(addr, timeoutMillis, responseFuture.getCause()));
+                    } else {
+                        future.completeExceptionally(new RemotingException(request.toString(), responseFuture.getCause()));
+                    }
+                }
+            });
+        } catch (Exception e) {
```

Comment:
Suggest catch Throwable instead of Exception.

Reply:
fixed